### ISSUE TITLE
yunyang，关于没有xy_loss 和wh_loss的问题

### ISSUE THREAD

**Issue by hry8310** [YunYang1994/tensorflow-yolov3#277]
您好！
       我使用keral的原版对比了你的实现，其中loss 函数差别主要是：你使用了giou_loss 代替了原版的xy_loss 和 wh_loss？如果是这样，原版的是不是也可以做出这样的改动，我尝试了一下。不行。能不能说明一下呢：比如说明一下。在原版是否可以使用giou（或者iou）损失代替 xy_loss 和 wh_loss

**Comment by hry8310** [comment]
yunyang，您好！我看到你loss一开始就是从原版类似的，后面才逐步优化的，能告诉我们有哪些差异吗？如果在原版上修改为采用giou代替 xy\wh。需要注意哪些东西。涉及修改的地方会不会很多，比如输入的shape

**Comment by YunYang1994** [comment]
Keras 原版？sorry，我没看过这个。

**Comment by hry8310** [comment]
谢谢你的回复。sorry，是 keras-yolo3-detection 这个。代码：https://github.com/SpikeKing/keras-yolo3-detection 。我看你有关yolo的wiki 也是是引用类似的代码。你早期的版本还是有 xy\wh 的loss，5月版的提交才优化掉了（优化逻辑能分享一个wiki就感谢大神了）。另外，请教一下：你代码中的 loss_layer(self, conv, pred, label, bboxes, anchors, stride): 中的 pred，label 是以原始像素为单位的xywh。还是以gird加偏移的格式的。还有bboxes 表达的是什么意义的boxes

**Comment by shixiaochong** [comment]
=> EPOCH         70 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:123.3714  loss_class: 0.0000
=> EPOCH         71 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:120.0964  loss_class: 0.0000
=> EPOCH         72 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:117.3832  loss_class: 0.0000
=> EPOCH         73 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:121.9322  loss_class: 0.0000
=> EPOCH         74 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:113.0575  loss_class: 0.0000
=> EPOCH         75 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:112.4483  loss_class: 0.0000
=> EPOCH         76 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:110.6726  loss_class: 0.0000
=> EPOCH         77 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:110.7863  loss_class: 0.0000
=> EPOCH         78 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:106.6857  loss_class: 0.0000
=> EPOCH         79 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:105.4851  loss_class: 0.0000
=> EPOCH         80 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:104.9139  loss_class: 0.0000
=> EPOCH         81 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:105.3739  loss_class: 0.0000
=> EPOCH         82 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:101.7272  loss_class: 0.0000
=> EPOCH         83 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:100.8565  loss_class: 0.0000
=> EPOCH         84 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:99.9467   loss_class: 0.0000
=> EPOCH         85 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:98.9660   loss_class: 0.0000
=> EPOCH         86 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:97.0807   loss_class: 0.0000
=> EPOCH         87 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:94.1452   loss_class: 0.0000
=> EPOCH         88 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:94.1438   loss_class: 0.0000
=> EPOCH         89 [TRAIN]:    loss_xy: 0.0000         loss_wh: 0.0000         loss_conf:96.0400   loss_class: 0.0000

**Comment by zhouxianming** [comment]
> 谢谢你的回复。sorry，是 keras-yolo3-detection 这个。代码：https://github.com/SpikeKing/keras-yolo3-detection 。我看你有关yolo的wiki 也是是引用类似的代码。你早期的版本还是有 xy\wh 的loss，5月版的提交才优化掉了（优化逻辑能分享一个wiki就感谢大神了）。另外，请教一下：你代码中的 loss_layer(self, conv, pred, label, bboxes, anchors, stride): 中的 pred，label 是以原始像素为单位的xywh。还是以gird加偏移的格式的。还有bboxes 表达的是什么意义的boxes

@hry8310 
你好，请问你知道了吗为什么预测loss的时候用giou代替 xy\wh？
可否请教一下，谢谢

**Comment by hry8310** [comment]
@zhouxianming 
可能是因为对比正确率的吧。这两种方式逻辑都没有问题。训练起来，loss都能收敛。

**Comment by hry8310** [comment]
@zhouxianming  顺便说一句，其实GIOU是可以使用IOU代替的。只是 GIOU理论上准确度更高而已。我试过用IOU也是能够收敛的
